crates/blockchain/payload.rs

```diff
@@ -289,27 +291,30 @@ pub fn fill_transactions(context: &mut PayloadBuildContext) -> Result<(), ChainE
             txs.pop();
             continue;
         }
-        // Pull transaction from the mempool
+
         // TODO: maybe fetch hash too when filtering mempool so we don't have to compute it here (we can do this in the same refactor as adding timestamp)
         let tx_hash = head_tx.tx.compute_hash();
-        mempool::remove_transaction(tx_hash, context.store())?;
 
         // Check wether the tx is replay-protected
         if head_tx.tx.protected() && !chain_config.is_eip155_activated(context.block_number()) {
             // Ignore replay protected tx & all txs from the sender
+            // Pull transaction from the mempool
             debug!("Ignoring replay-protected transaction: {}", tx_hash);
             txs.pop();
+            mempool::remove_transaction(tx_hash, context.store())?;
```

Comment:
should we do this here or when adding a block?

Reply:
I was looking at what geth code but I misinterpreted the comment, it looks like transactions are not removed here, but I couldn't find where the mempool is cleaned via quick codebase search, Ill create an issue to investigate this further. In the meantime should we leave this as is?

Reply:
(#796)

Reply:
lol I had just created https://github.com/lambdaclass/ethereum_rust/issues/797 too